### PR TITLE
[vividus] Allow to set fail fast at batch level

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: vividus-tests/src/main/resources/story/system/mobile_app/MobitruHealthCheckMobileApp.story [hunk 1]
```diff
@@ -16,3 +16,6 @@ Examples:
 Scenario: Validate application start
 When I wait until element located by `accessibilityId(test-Username)` appears
 When I take screenshot
+When I type `text` in field located `locator`
+When I press key key
+When I type `text` in field located `locator` and keep keyboard opened
```
Comment (valfirst) [inline vividus-tests/src/main/resources/story/system/mobile_app/MobitruHealthCheckMobileApp.story:21]: is it expected change?

File: vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java [hunk 2]
```diff
@@ -130,4 +152,24 @@ private List<String> convertToList(String list)
                      .map(String::strip)
                      .collect(Collectors.toList());
     }
+
+    private static final class Story
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:156]:
```suggestion
    private static final class StoryExecutionConfiguration
```

File: vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java [hunk 3]
```diff
@@ -130,4 +152,24 @@ private List<String> convertToList(String list)
...
+    {
+        private boolean failFast;
+
+        private void setFailFast(boolean failFast)
+        {
+            this.failFast = failFast;
+        }
+    }
+
+    private static final class Scenario
```
Comment (valfirst) [inline vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:166]:
```suggestion
    private static final class ScenarioExecutionConfiguration
```

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

File: vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java [hunk 4]
```diff
@@ -130,4 +152,24 @@ private List<String> convertToList(String list)
...
+
+    private static final class ScenarioExecutionConfiguration
+    {
+        private boolean failFast;
+
+        private void setFailFast(boolean failFast)
+        {
+            this.failFast = failFast;
+        }
+    }
```
Comment (uarlouski) [inline vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:174]: minor question, do we need both StoryExecutionConfiguration and ScenarioExecutionConfiguration? it look like we could use one class with generic name

**Review by uarlouski — APPROVED** [review]

File: vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java (hunk 4 above)
Comment (ikalinin1) [reply vividus-engine/src/main/java/org/vividus/batch/BatchConfiguration.java:174]: Based on the issue https://github.com/vividus-framework/vividus/issues/3416 this value will be inserted into the story config

**Pull request merged**